includes/Modules/Analytics_4/Synchronize_Property.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Analytics_4\Synchronize_Property
+ *
+ * @package   Google\Site_Kit\Modules\Analytics_4
+ * @copyright 2023 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Analytics_4;
+
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Modules\Analytics_4;
+use Google\Site_Kit_Dependencies\Google\Service\GoogleAnalyticsAdmin\GoogleAnalyticsAdminV1betaProperty;
+
+/**
+ * The base class for Synchronizing the Analytics 4 property.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Synchronize_Property {
+
+	const CRON_SYNCHRONIZE_PROPERTY = 'googlesitekit_cron_synchronize_property_data';
+
+	/**
+	 * Analytics_4 instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Analytics_4
+	 */
+	protected $analytics_4;
+
+	/**
+	 * User_Options instance.
+	 *
+	 * @since n.e.x.t
+	 * @var User_Options
+	 */
+	protected $user_options;
+
+	/**
+	 * User_Options instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Authentication
+	 */
+	protected $authentication;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Analytics_4    $analytics_4 Analytics 4 instance.
+	 * @param User_Options   $user_options User_Options instance.
+	 * @param Authentication $authentication Authentication instance.
+	 */
+	public function __construct( Analytics_4 $analytics_4, User_Options $user_options, Authentication $authentication ) {
+		$this->analytics_4    = $analytics_4;
+		$this->user_options   = $user_options;
+		$this->authentication = $authentication;
+	}
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+
+		add_action(
+			self::CRON_SYNCHRONIZE_PROPERTY,
+			function() {
+				$owner_id         = $this->analytics_4->get_owner_id();
+				$original_user_id = $this->user_options->get_user_id();
+
+				$this->user_options->switch_user( $owner_id );
+
+				if ( $this->authentication->is_authenticated() ) {
+					$property = $this->retrieve_property();
+
+					$this->synchronize_property_create_time( $property );
+				}
+
+				$this->user_options->switch_user( $original_user_id );
+			}
```

Comment:
Please move the body to a method rather than having it inline in `register` (a pattern used elsewhere which we should get away from)